docs/guide/component-testing.md

```diff
@@ -0,0 +1,402 @@
+
+Install the Vue testing package:
+
+```bash
+npm install --save-dev vitest-browser-vue
+```
+
+```vue
+<!-- HelloWorld.vue -->
+<script setup lang="ts">
+defineProps<{
+  name: string
+}>()
+</script>
+
+<template>
+  <div>
+    <h1>Hello {{ name }}!</h1>
+  </div>
+</template>
+```
+
+```ts
+// HelloWorld.test.ts
+import { expect, test } from 'vitest'
+import { render } from 'vitest-browser-vue'
+import HelloWorld from './HelloWorld.vue'
+
+test('renders name', async () => {
+  const { getByText } = render(HelloWorld, {
+    props: { name: 'Vitest' },
+  })
+  await expect.element(getByText('Hello Vitest!')).toBeInTheDocument()
+})
+```
+
+#### React Components
+
+Install the React testing package:
+
+```bash
+npm install --save-dev vitest-browser-react
+```
+
+```tsx
+// HelloWorld.tsx
+export default function HelloWorld({ name }: { name: string }) {
+  return (
+    <div>
+      <h1>
+        Hello
+        {' '}
+        {name}
+        !
+      </h1>
+    </div>
+  )
+}
+```
+
+```tsx
+// HelloWorld.test.tsx
+import { expect, test } from 'vitest'
+import { render } from 'vitest-browser-react'
+import HelloWorld from './HelloWorld.tsx'
+
+test('renders name', async () => {
+  const { getByText } = render(<HelloWorld name="Vitest" />)
+  await expect.element(getByText('Hello Vitest!')).toBeInTheDocument()
+})
+```
+
+#### Svelte Components
+
+Install the Svelte testing package:
+
+```bash
+npm install --save-dev vitest-browser-svelte
+```
+
+```svelte
+<!-- HelloWorld.svelte -->
+<script lang="ts">
+  export let name: string
+</script>
+
+<h1>Hello {name}!</h1>
+```
+
+```ts
+// HelloWorld.test.ts
+import { expect, test } from 'vitest'
+import { render } from 'vitest-browser-svelte'
+import HelloWorld from './HelloWorld.svelte'
+
+test('renders name', async () => {
+  const { getByText } = render(HelloWorld, { name: 'Vitest' })
+  await expect.element(getByText('Hello Vitest!')).toBeInTheDocument()
+})
+```
+
+#### Lit Components
+
+Install the community Lit testing package:
+
+```bash
+npm install --save-dev vitest-browser-lit
+```
+
+```ts
+// HelloWorld.ts
+import { html, LitElement } from 'lit'
+import { customElement, property } from 'lit/decorators.js'
+
+@customElement('hello-world')
+export class HelloWorld extends LitElement {
+  @property({ type: String })
+  name = 'World'
+
+  render() {
+    return html`<h1>Hello ${this.name}!</h1>`
+  }
+}
+
+declare global {
+  interface HTMLElementTagNameMap {
+    'hello-world': HelloWorld
+  }
+}
+```
+
+```ts
+// HelloWorld.test.ts
+import { expect, test } from 'vitest'
+import { render } from 'vitest-browser-lit'
+import { html } from 'lit'
+import './HelloWorld.js'
+
+test('renders name', async () => {
+  const screen = render(html`<hello-world name="Vitest"></hello-world>`)
+  const element = screen.getByText('Hello Vitest!')
+  await expect.element(element).toBeInTheDocument()
+})
+```
+
+## User Interactions
+
+Test user interactions using Vitest's locator methods:
```

Comment:
Would be nice to have the link here to `/guide/browser/interactivity-api` already